agent/agent.go

```diff
@@ -348,7 +345,7 @@ type processNextResponse struct {
 	tc            *taskContext
 }
 
-func (a *Agent) processNextTask(nt *apimodels.NextTaskResponse, ctx context.Context, tc *taskContext, needPostGroup bool) (processNextResponse, error) {
+func (a *Agent) processNextTask(ctx context.Context, tc *taskContext, nt *apimodels.NextTaskResponse, needPostGroup bool) (processNextResponse, error) {
```

Comment:
nit: I would make the next task response the 2nd argument since it's the main thing being processed in the function, and the taskContext and needPostGroup are just data retained from the previous task that ran.